client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClient.java

```diff
@@ -249,8 +249,29 @@ protected static final void setApplicationProperties(final Message msg, final Ma
             receiver.setAutoAccept(true);
             receiver.setQoS(qos);
             receiver.setPrefetch(clientConfig.getInitialCredits());
-            receiver.handler(messageHandler);
-            receiver.openHandler(result.completer());
+            receiver.handler((delivery, message) -> {
+                messageHandler.handle(delivery, message);
+                if (LOG.isTraceEnabled()) {
+                    int remainingCredits = receiver.getCredit() - receiver.getQueued();
+                    LOG.trace("handling message [remotely settled: {}, queued messages: {}, remaining credit: {}]", delivery.remotelySettled(), receiver.getQueued(), remainingCredits);
+                }
+            });
+            receiver.openHandler(openAttach -> {
+                if(openAttach.failed()) {
+                    result.fail(openAttach.cause());
+                    LOG.debug("receiver open attach failed [{}] by peer [{}]: {}", receiver.getRemoteSource(), con.getRemoteContainer(), openAttach.cause().getMessage());
```

Comment:
IMHO this log statement should be placed *before* the line where the result is failed so that in the log this statement shows up before any other statements logged by the result handler.